### PR TITLE
refactor: remove `PackValue` and `Convert` functions from src/codec/field_codec.h

### LINKED ISSUE #1992: Remove `PackValue` and `Convert` functions from src/codec/field_codec.h
- remove `PackValue` and `Convert` functions from src/codec/field_codec.h
- remove GetXXX functions from src/codec/field_codec.h
- remote `#include "codec/memcomparable_format.h"`  from src/codec/field_codec.h

delete the following files:
- src/codec/memcomparable_format.h
- src/codec/memcomparable_format_test.cc
- src/codec/field_codec_test.cc

### PR DESCRIPTION
**Pull request by frazie**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
fixed issue #1992 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: src/codec/field_codec.h [hunk 1]
```diff
@@ -38,216 +37,10 @@ using ::openmldb::type::DataType;
 /**
  *  encode part
```
Comment (dl239) [inline src/codec/field_codec.h]: delete the annotation as well

Comment (frazie) [reply src/codec/field_codec.h]: done!

File: src/codec/field_codec.h [hunk 2]
```diff
@@ -27,227 +27,16 @@
 #include "base/glog_wapper.h"
 #include "base/strings.h"
 #include "boost/lexical_cast.hpp"
-#include "codec/memcomparable_format.h"
 #include "proto/type.pb.h"
 #include "sdk/base.h"
 
-namespace openmldb {
-namespace codec {
```
Comment (dl239) [inline src/codec/field_codec.h:35]: do not delete the namespace declaration
https://github.com/4paradigm/OpenMLDB/runs/6996479985?check_suite_focus=true#step:7:187

Comment (frazie) [reply src/codec/field_codec.h:35]: error fixed, sorry for that mistaken deletion

File: src/codec/field_codec.h (hunk 1 above)
Comment (dl239) [reply src/codec/field_codec.h]: please delete the following annotation
https://github.com/4paradigm/OpenMLDB/blob/cfe800638f5443465a078da07d86dbd962682e40/src/codec/field_codec.h#L37-L43

Comment (frazie) [reply src/codec/field_codec.h]: > please delete the following annotation
> 
> https://github.com/4paradigm/OpenMLDB/blob/cfe800638f5443465a078da07d86dbd962682e40/src/codec/field_codec.h#L37-L43

done

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**